M1FExplained/Chapter10/Exercises/Exercise08.lean

```diff
@@ -0,0 +1,76 @@
+import Mathlib.Tactic
+
+namespace Chapter10.Exercise08
+
+/-
+Let n ≥ 2 be an integer. Prove that n is prime if and only if for every integer a, 
+either gcd(a, n) = 1 or n ∣ a. 
+-/
+
+lemma exercise08 (n : ℤ) (hn : 2 ≤ n) : Prime n ↔ ∀ (a : ℤ), Int.gcd a n = 1 ∨ n ∣ a := by
+  -- Since 2 ≤ n, it is a natural number, we can rewrite this in the local context.
+  rw [show n = n.natAbs by 
+      rw [Int.coe_natAbs] 
+      exact Eq.symm (abs_of_nonneg (show 0 ≤ n by linarith))] at *
+  -- N is the coercion on n to the natural numbers.
+  set N := n.natAbs
+  constructor 
+  <;> intro h 
+  <;> rw [← Nat.prime_iff_prime_int] at *
+  -- → direction. 
+  · intro a
+  -- Condition on whether N divides |a|.
+    apply Or.elim (Classical.em (N ∣ a.natAbs))
+    <;> intro h'
+  -- If N ∣ |a|, then trivially N ∣ |a|.
+    · right
+      exact Iff.mpr Int.ofNat_dvd_left h'
+  -- If ¬ N ∣ |a|, then since N is prime, gcd(N, a) = 1. This is proven by contradiction.
+    · left
+      by_contra h''
+      set M := Int.gcd a N with hM
+  -- Condition on wether a = 0 or not.
```

Comment:
You never use condition `ha` in the "not zero" case, so you can just not do this case split and use the second argument to do the a=0 case as well.